### PR TITLE
1.21 key prefix

### PR DESCRIPTION
**Pull request by rnicoll**
Update the address and public/private key prefixes used on main and testnet to Dogecoin values (as in, why Dogecoin addresses start with "D" where Bitcoin uses "1").

### CODE REVIEW

**Comment by braxton6** [comment]
Can someone explain whats going on and how does this relate to helping develop dogecoin? 

Thank you

File: test/functional/wallet_upgradewallet.py [hunk 1]
```diff
@@ -1,354 +0,0 @@
-#!/usr/bin/env python3
```
Comment (patricklodder) [inline test/functional/wallet_upgradewallet.py:1]: also better to keep the file?

**Review by patricklodder — COMMENTED** [review]
Have left inline comments for the most part, and one additional generic question: some of the segwit/pbst encoded private keys seem to have changed but I am unclear why. Are there any prefixes or magic numbers that we have to define?

File: test/functional/feature_backwards_compatibility.py [hunk 2]
```diff
@@ -1,426 +0,0 @@
-#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (rnicoll) [reply test/functional/feature_backwards_compatibility.py:1]: I had a bit of an argument with the test runner about this. It seems if you have an unutilised test the test runner warns which breaks Travis. Honestly, didn't try _that_ hard, might be worth another go to be sure.

**Comment by rnicoll** [comment]
> Have left inline comments for the most part, and one additional generic question: some of the segwit/pbst encoded private keys seem to have changed but I am unclear why. Are there any prefixes or magic numbers that we have to define?

Private key prefix changes on https://github.com/dogecoin/dogecoin/pull/1710/files#diff-ff53e63501a5e89fd650b378c9708274df8ad5d38fcffa6c64be417c4d438b6dR133 . All keys _should_ be identical except for prefix and hash, do let me know if anything weird snuck through though.

**Comment by patricklodder** [comment]
> Can someone explain whats going on and how does this relate to helping develop dogecoin?

For the next major release, we're building off of bitcoin 0.21, so we basically take all Dogecoin specific functionality and re-apply that onto that Bitcoin Core code, so that we benefit from all the awesome work done for Bitcoin and don't do double work. You can identify these pull requests by having the milestone tag "1.21"

File: src/test/data/key_io_invalid.json [hunk 3]
```diff
@@ -156,25 +156,25 @@
...
     [
-        "BC1QR508D6QEJXTDG4Y5R3ZARVARYV98GJ9P"
+        "DG1QR508D6QEJXTDG4Y5R3ZARVARYVGU055V"
     ],
     [
         "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3q0sL5k7"
     ],
     [
-        "bc1zw508d6qejxtdg4y5r3zarvaryvqyzf3du"
+        "dg1zw508d6qejxtdg4y5r3zarvaryvqndalvf"
```
Comment (patricklodder) [inline src/test/data/key_io_invalid.json:174]: This one looks wrong to me. it will fail for the wrong reason?

Comment (rnicoll) [reply src/test/data/key_io_invalid.json]: Good catch, fixed

**Review by patricklodder — APPROVED** [review]
this looks good to me now.

**Pull request merged**